### PR TITLE
Create code school card component

### LINKED ISSUE #288: Create Code School Card component
# Feature request
**Is your feature request related to a problem? Please describe.**
We have new mocks for `/code_schools` which requires a redo of our Code Schools Card component.

**Describe the solution you'd like**
Emulate the code school card interface outlined in [this mock](https://projects.invisionapp.com/d/main#/console/15378934/323736695/preview)

**Additional context**
- Anywhere where the light blue (primary) color exists, please use secondary instead. After review with design, we decided to dark up the UI of the page, but elected to not update the mock.
- Ignore the custom scroll bar visible in mock when looking at back of multi-location school card - use browser default scroll.
- You can find the SVG assets I'd like used in this unmerged pull request: https://github.com/OperationCode/front-end/pull/137
- Use the same code school logo on both sides of the card. We will eventually make them `.png` files so that they have a transparent background.
- While the mock attempts to communicate clicking on a location in the location list on the back of the card opens a new page, it will actually just be plain text.
- Ignore the housing icon. Need to make a separate issue to get it in when https://github.com/OperationCode/operationcode_backend/pull/415 is merged.
- When the view width is too small, stack the buttons on top of each other vertically
- Do your best to get a min-width which keeps the icons spanning only across two rows - feel free to connect the Badges to media queries that'll reduce their size.

**Components You Will Use**
Card
Button (for locations button)
LinkButton (for website button)
Badge

### PR DESCRIPTION
**Pull request by markchernov**
# Description of changes

Created SchoolCard
# Issue Resolved

Fixes #288

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 17daa98c8b28db730bbc22bf4f142a7b619e1f29

https://deploy-preview-300--operation-code-storybook.netlify.com

File: components/Cards/SchoolCard/FrontSchoolCard.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            </span>
+          )}
+        </div>
+
+        <hr className={styles.divider} />
+
+        <div className={styles.detailsRow}>
+          {/* GI Bill */}
+          {props.locations.some(location => location.va_accepted) ? (
+            <figure className={`${styles.detailItem} ${styles.active}`}>
```
Comment (kylemh) [inline components/Cards/SchoolCard/FrontSchoolCard.js]: Let's change the `<Badge>` component to use `<figure>` and `<figcaption>` and then use the Badge component here. We can also drop `aria-hidden="true"` from all of these, since they're added automatically now via SVG Loader config.

File: components/Cards/SchoolCard/__tests__/FrontSchoolCard.test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    va_accepted: true,
+    address1: '825 Battery Street',
+    address2: '3rd Floor',
+    city: 'San Francisco',
+    zip: '94111',
+  },
+];
+
+describe('FrontSchoolCard', () => {
+  it('should render with required props', () => {
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/FrontSchoolCard.test.js]: I'd like to see tests similar to `Button.test.js` line 65 "call props.onClick when button is clicked" but for the callback on both front and back card tests.

File: components/Cards/SchoolCard/__tests__/SchoolCard.test.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    va_accepted: true,
+    address1: '825 Battery Street',
+    address2: '3rd Floor',
+    city: 'San Francisco',
+    zip: '94111',
+  },
+];
+
+describe('SchoolCard', () => {
+  it('should render with required props', () => {
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/SchoolCard.test.js]: I'd like to see tests similar to `Button.test.js` line 65 "call props.onClick when button is clicked" but for the callback on both front and back card tests. The only difference is it should check for the state change too. This time, however, check for the relevant state variable to have been changed after.

File: components/_common_/CloseButton/CloseButton.css [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  right: 1rem;
+  background: var(--slate);
+  border-radius: 50%;
+  border: none;
+  position: absolute;
+}
+
+.CloseButton:disabled {
+  cursor: not-allowed;
+  background-color: var(--slateLight);
```
Comment (kylemh) [inline components/_common_/CloseButton/CloseButton.css]: I've removed the CloseButton because it didn't align with the new theme. I'll see about making a new close button with the same API that we can merge into this PR.

File: components/Badge/Badge.js [hunk 5]
```diff
@@ -23,13 +23,13 @@ function Badge({ className, icon, isImageFirst, label }) {
     <div className={classNames(styles.Badge, className)}>
       {isImageFirst ? (
         <>
-          <div className={styles.iconContainer}>{icon}</div>
-          <span className={styles.label}>{label}</span>
+          <figure className={styles.iconContainer}>{icon}</figure>
```
Comment (kylemh) [inline components/Badge/Badge.js]: so `<figcaption>` is meant to be a child of `<figure`, not a sibling.

Basically, I'd expect a figure instead of a React.Fragment, and then the div can stay behind. You did the figcaption in the right spot tho!

File: components/Badge/Badge.css [hunk 6]
```diff
@@ -3,18 +3,18 @@
   flex-direction: column;
   justify-content: flex-start;
   align-items: center;
-  margin: 1rem;
+  margin: 2rem;
```
Comment (kylemh) [inline components/Badge/Badge.css]: Badge exists elsewhere in the app, so these changes affect those usages. Let's leave the stylesheet here untouched. Instead, we can override the Badge instances in the code school card with a className.

`<Badge className={styles.codeSchoolCardIcon} ...`

File: components/Cards/SchoolCard/__strories__/SchoolCardBack.stories.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
```
Comment (kylemh) [inline components/Cards/SchoolCard/__strories__/SchoolCardBack.stories.js]: It might be better to drop the stories for the front and back school card, while just making sure the SchoolCard itself has a fleshed out story

File: components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    address2: '3rd Floor',
+    city: 'San Francisco',
+    zip: '94111',
+  },
+];
+
+describe('BackSchoolCard', () => {
+  it('should render with required props', () => {
+    createShallowSnapshotTest(
+      <BackSchoolCard cardFlipCallback={() => {}} locations={locations} logoSource="logoSource">
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js]: can you do `jest.fn()` instead of `() => {}`

File: components/Cards/SchoolCard/__tests__/SchoolCard.test.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        locations={locations}
+        logoSource="source"
+        name="school name"
+        website="website"
+      >
+        Test
+      </SchoolCard>,
+    );
+    const instance = wrapper.instance();
+    wrapper.setState({ isFrontOfCardShowing: true });
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/SchoolCard.test.js]: line 52 and 53 are unnecessary.

I'd also love if you tested content existing AND the state variable... So like check for the existence of content that exists only on the front card when `isFrontOfCardShowing` is true

Comment (markchernov) [reply components/Cards/SchoolCard/__tests__/SchoolCard.test.js]: Updated tests here

File: components/Badge/Badge.js (hunk 5 above)
Comment (markchernov) [reply components/Badge/Badge.js]: Refactored Badge

File: components/_common_/CloseButton/CloseButton.css (hunk 4 above)
Comment (markchernov) [reply components/_common_/CloseButton/CloseButton.css]: Close button still seems to be present

File: components/Cards/SchoolCard/__strories__/SchoolCardBack.stories.js (hunk 7 above)
Comment (markchernov) [reply components/Cards/SchoolCard/__strories__/SchoolCardBack.stories.js]: Never done these stories before, will take a look

File: components/_common_/CloseButton/CloseButton.css (hunk 4 above)
Comment (kylemh) [reply components/_common_/CloseButton/CloseButton.css]: I merged in an updated CloseButton 👍

File: components/Cards/SchoolCard/FrontSchoolCard.js [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+          </div>
+
+          {/* In-Person Education */}
+          <div
+            className={classNames(styles.detailItem, {
+              [styles.active]: props.isFullTime,
+              [styles.disabled]: !props.isFullTime,
+            })}
+          >
+            {' '}
```
Comment (kylemh) [inline components/Cards/SchoolCard/FrontSchoolCard.js]: I don't think you meant to have this here

File: components/Cards/SchoolCard/FrontSchoolCard.js [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+
+        <div className={styles.detailsRow}>
+          {/* Equipment Provided */}
+          <div
+            className={classNames(styles.detailItem, {
+              [styles.active]: props.hasHardwareIncluded,
+              [styles.disabled]: !props.hasHardwareIncluded,
+            })}
+          >
+            {' '}
```
Comment (kylemh) [inline components/Cards/SchoolCard/FrontSchoolCard.js]: I don't think you meant to have this here

File: components/Cards/SchoolCard/FrontSchoolCard.js [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+          </div>
+
+          {/* Housing Provided */}
+          <div
+            className={classNames(styles.detailItem, {
+              [styles.active]: props.hasHousing,
+              [styles.disabled]: !props.hasHousing,
+            })}
+          >
+            {' '}
```
Comment (kylemh) [inline components/Cards/SchoolCard/FrontSchoolCard.js]: I don't think you meant to have this here

File: components/Cards/SchoolCard/FrontSchoolCard.js [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+  render() {
+    const { props } = this;
+
+    return (
+      <>
+        <h5>{props.name}</h5>
+        <img src={props.logoSource} alt="" />
+
+        <div className={styles.interactionsContainer}>
+          <Button>
```
Comment (kylemh) [inline components/Cards/SchoolCard/FrontSchoolCard.js]: Let's drop `<Button><OutboundLink>...` here and use the new `<LinkButton>` component.

File: pages/code_schools.js [hunk 14]
```diff
@@ -46,7 +48,23 @@ export default class CodeSchools extends React.Component {
         </Section>
         <Section theme="gray" title="Schools" hasHeadingLines={false}>
           {state.schools.map(school => (
-            <div key={`${Math.random()}`}>{school.name}</div>
+            <div key={`${Math.random()}`}>
```
Comment (kylemh) [inline pages/code_schools.js]: I think you'll learn something new when reading [this](https://stackoverflow.com/questions/29808636/when-giving-unique-keys-to-components-is-it-okay-to-use-math-random-for-gener).

I recommend using `school.name` as the key.

File: components/Cards/SchoolCard/__strories__/SchoolCardBack.stories.js (hunk 7 above)
Comment (kylemh) [reply components/Cards/SchoolCard/__strories__/SchoolCardBack.stories.js]: Ah, cool.

Okay, to clarify, I was suggesting deleting the `__stories__` folder for the front and back card.

You currently have code in `SchoolCardFront.stories.js` which I would consider perfect for `components/Cards/SchoolCard/__stories__/SchoolCard.stories.js`. So just have that one Story file, and you're good to go.

Also, really quick note: you've got a typo in the folder path: `__strories__` should be `__stories__`

File: components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js [hunk 15]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';
+import BackSchoolCard from '../BackSchoolCard';
+
+const locations = [
+  {
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js]: since the BackSchoolCard wouldn't normally be renderable with just one location, let's add a second location to the test data. I suggest doing multiple locations in each of the tests. Consider doing 
```
[ { ...locations[0] } ]
```
if you want to test a component with a list of 1 (potentially valuable in the front school card).

File: components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+];
+
+describe('BackSchoolCard', () => {
+  it('should render with required props', () => {
+    createShallowSnapshotTest(
+      <BackSchoolCard cardFlipCallback={jest.fn()} locations={locations} logoSource="logoSource">
+        <p>Testing!</p>
+      </BackSchoolCard>,
+    );
+  });
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js]: Add the following test for me:
```javascript
it('should fire flip callback when close button is pressed', () => {
    const onClickMock = jest.fn();
    const wrapper = mount(
      <BackSchoolCard
        cardFlipCallback={onClickMock}
        locations={locations}
        logoSource="logoSource"
      />,
    );

    wrapper.find('button').simulate('click');
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
```

This is an integration test ensuring that the CloseButton does what it's supposed to within the BackSchoolCard.

File: components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+    city: 'San Francisco',
+    zip: '94111',
+  },
+];
+
+describe('BackSchoolCard', () => {
+  it('should render with required props', () => {
+    createShallowSnapshotTest(
+      <BackSchoolCard cardFlipCallback={jest.fn()} locations={locations} logoSource="logoSource">
+        <p>Testing!</p>
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js]: so nested elements within a component would be considered the same thing as the `children` prop.

While we provide it as a default for the `createComponent` script, it's unnecessary here (or in any of your school card-related tests since none of those components deal with the `children` prop.)

See below for an example

File: components/Cards/SchoolCard/__tests__/FrontSchoolCard.test.js (hunk 2 above)
Comment (kylemh) [reply components/Cards/SchoolCard/__tests__/FrontSchoolCard.test.js]: I outlined an example test in the BackSchoolCard.test.js file 🕺

File: components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js [hunk 18]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';
+import BackSchoolCard from '../BackSchoolCard';
+
+const locations = [
+  {
+    va_accepted: true,
+    address1: '825 Battery Street',
+    address2: '3rd Floor',
+    city: 'San Francisco',
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js]: Let's provide `state` if city is provided. This one requires a bit of context, but basically the back-end will either not provide us EITHER of city or state, or they provide both. Never one separately. It's tough to communicate that with prop-types, but make sure the test data reflects such.

File: components/Cards/SchoolCard/__tests__/SchoolCard.test.js [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+    address2: '3rd Floor',
+    city: 'San Francisco',
+    zip: '94111',
+  },
+];
+
+describe('SchoolCard', () => {
+  let wrapper;
+  beforeEach(() => {
+    wrapper = shallow(
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/SchoolCard.test.js]: I like you trying to remove duplicate code with the before each!

You can take it a step farther by defining 

```
const componentInstance = <SchoolCard {...etc} />
```

and using it in `wrapper = shallow(componentInstance)` PLUS `createShallowSnapshotTest(componentInstance)`

File: components/Cards/SchoolCard/__tests__/SchoolCard.test.js [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+        logoSource="logoSource.com"
+        name="school name"
+        website="website.com"
+      >
+        <p>Testing!</p>
+      </SchoolCard>,
+    );
+  });
+
+  it('call to showBackOfCard/showFrontOfTheCard flips isFrontOfCardShowing', () => {
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/SchoolCard.test.js]: Let's name this a little differently. I suggest: `should flip card when locations button is clicked`

Additionally, instead of just testing that state changes when the function is called, what if we tested the changes after simulating the actual interaction that calls the function? (See CloseButton tests)

File: components/Cards/SchoolCard/__tests__/SchoolCard.test.js [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+    instance.showBackOfCard();
+    expect(wrapper.state('isFrontOfCardShowing')).toBe(false);
+    instance.showFrontOfCard();
+    expect(wrapper.state('isFrontOfCardShowing')).toBe(true);
+  });
+
+  it('renders the See Locations Buttom', () => {
+    expect(wrapper.state('isFrontOfCardShowing')).toBe(true);
+    expect(wrapper.find('See Locations')).toBeTruthy();
+  });
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/SchoolCard.test.js]: Good idea to test this the existence of the "See Locations" button. 

We always want to think about "what are we really testing?"

Looking at the logic of the component, this button will only render if there are multiple locations passed to the component. Let's test that!

```javascript
it('should render the "See Locations" button when multiple locations exist', () => {
  // ensure wrapper has multiple locations at this point
  expect(wrapper.prop('locations')).toBeGreaterThan(1);
  expect(wrapper.find('button').filterWhere(node => node.text() === 'See Locations')).toExist();
});
```

File: components/Cards/SchoolCard/__tests__/SchoolCard.test.js [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+
+  it('renders the text when flipped', () => {
+    const instance = wrapper.instance();
+    instance.showBackOfCard();
+    expect(wrapper.state('isFrontOfCardShowing')).toBe(false);
+    expect(wrapper.find('denotes a location that accepts the GI Bill')).toBeTruthy();
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/SchoolCard.test.js]: This is great! You're clearly testing that the card flips properly. I just wanna suggest some tweaks here, but the testing logic is 👌 

```
it('renders back of school card when "See Locations" button is clicked', () => {
  const textUniqueToBackSchoolCard = 'denotes a location that accepts the GI Bill';

  expect(wrapper.find(textUniqueToBackSchoolCard)).not.toExist();
  wrapper.find('button').filterWhere(node => node.text() === 'See Locations').simulate('click');
  expect(wrapper.find(textUniqueToBackSchoolCard)).toExist();
});
```

File: components/Cards/SchoolCard/BackSchoolCard.js [hunk 23]
```diff
@@ -29,7 +30,7 @@ function BackSchoolCard({ cardFlipCallback, locations, logoSource }) {
 
       <ul className={styles.locations}>
         {locations.map(location => (
-          <li>
+          <li id={logoSource}>
```
Comment (kylemh) [inline components/Cards/SchoolCard/BackSchoolCard.js]: I think you meant `key` here

File: components/Cards/SchoolCard/SchoolCard.js [hunk 24]
```diff
@@ -50,6 +50,7 @@ class SchoolCard extends Component {
             cardFlipCallback={this.showFrontOfCard}
             locations={props.locations}
             logoSource={props.logoSource}
+            alt={`${props.schoolName} logo`}
```
Comment (kylemh) [inline components/Cards/SchoolCard/SchoolCard.js]: what if you generated the alt in the backschoolcard, but just passed school name here instead?

It's a bit of a disconnect for me to have alt defined here, when I don't even know about there being an image from here (in the code).

File: components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js [hunk 25]
```diff
@@ -10,14 +11,40 @@ const locations = [
     city: 'San Francisco',
     zip: '94111',
   },
+  {
+    va_accepted: true,
+    address1: '825 Battery Street',
+    address2: '3rd Floor',
+    city: 'San Francisco',
+    zip: '94111',
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js]: still need state here, and I'd use different data in the second object - you can make it up! have fun with it.

(same with front card tests)

File: components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js [hunk 26]
```diff
@@ -10,14 +11,40 @@ const locations = [
...
 describe('BackSchoolCard', () => {
   it('should render with required props', () => {
     createShallowSnapshotTest(
-      <BackSchoolCard cardFlipCallback={jest.fn()} locations={locations} logoSource="logoSource">
+      <BackSchoolCard
+        cardFlipCallback={jest.fn()}
+        locations={locations}
+        logoSource="logoSource"
+        alt="alt"
+      >
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js]: Idk if you were able to see this since it's on the left side of the diff viewer, but 👀 
https://github.com/OperationCode/front-end/pull/300/files/483089c0a80113786c0b08237adb6fdb9ea8f3d5..a54eed8aabc5c7c89bab51df75f175872f1a8c51#r236574017

File: components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js [hunk 27]
```diff
@@ -10,14 +11,40 @@ const locations = [
...
         <p>Testing!</p>
       </BackSchoolCard>,
     );
   });
+
+  it('should call callback function', () => {
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/BackSchoolCard.test.js]: when button is clicked*

File: components/Cards/SchoolCard/SchoolCard.js [hunk 28]
```diff
@@ -50,7 +50,7 @@ class SchoolCard extends Component {
             cardFlipCallback={this.showFrontOfCard}
             locations={props.locations}
             logoSource={props.logoSource}
-            alt={`${props.schoolName} logo`}
+            alt={props.name}
```
Comment (kylemh) [inline components/Cards/SchoolCard/SchoolCard.js]: what if we just pass this as schoolName?

line 24 in backschoolcard would then be:
``

File: components/Cards/SchoolCard/SchoolCard.css [hunk 29]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+.SchoolCard > img {
+  height: 150px;
+  margin: 1rem;
+}
+
+.backCard {
+  background-color: var(--secondary);
+  color: #ffffff;
```
Comment (kylemh) [inline components/Cards/SchoolCard/SchoolCard.css]: var(--white)

File: pages/code_schools.js [hunk 30]
```diff
@@ -51,7 +51,7 @@ export default class CodeSchools extends React.Component {
             <div key={`${school.name}`}>
               <SchoolCard
                 hasHardwareIncluded={school.hardware_included}
-                hasHousing={school.has_housing}
+                hasHousing={school.has_housing ? school.has_housing : false}
```
Comment (kylemh) [inline pages/code_schools.js]: You can revert this change. We're trusting the API to give us true or false in `has_housing`

Comment (markchernov) [reply pages/code_schools.js]: Actually some of the responses were missing the field so I was seeing some undefined errors in console

Comment (markchernov) [reply pages/code_schools.js]: But let me change it back real quick

Comment (kylemh) [reply pages/code_schools.js]: Yes, `has_housing` is actually not in the response. SchoolCard.propTypes should list `hasHousing` as not required - no?

If it doesn't that'll be the fix.

File: components/Cards/SchoolCard/BackSchoolCard.js [hunk 31]
```diff
@@ -27,7 +27,7 @@ function BackSchoolCard({ cardFlipCallback, locations, logoSource, schoolName })
       <hr className={styles.divider} />
       <ul className={styles.locations}>
         {locations.map(location => (
-          <li key={`${location}`}>
+          <li key={`${JSON.stringify(location)}`}>
```
Comment (kylemh) [inline components/Cards/SchoolCard/BackSchoolCard.js]: Instead of this, you should be able to just do `location.name` since we only have one card per school. It's unique, but small.

File: components/Cards/SchoolCard/FrontSchoolCard.js [hunk 32]
```diff
@@ -32,6 +32,10 @@ export default class FrontSchoolCard extends Component {
     website: PropTypes.string.isRequired,
   };
 
+  static defaultProps = {
+    hasHousing: false,
```
Comment (kylemh) [inline components/Cards/SchoolCard/FrontSchoolCard.js]: 👏 sorry about that

File: components/Cards/SchoolCard/SchoolCard.js [hunk 33]
```diff
@@ -9,7 +9,7 @@ import styles from './SchoolCard.css';
 class SchoolCard extends Component {
   static propTypes = {
     hasHardwareIncluded: PropTypes.bool.isRequired,
-    hasHousing: PropTypes.bool.isRequired,
+    hasHousing: PropTypes.bool,
```
Comment (kylemh) [inline components/Cards/SchoolCard/SchoolCard.js]: sorry about that!

👏

File: components/Cards/SchoolCard/BackSchoolCard.js (hunk 31 above)
Comment (markchernov) [reply components/Cards/SchoolCard/BackSchoolCard.js]: I think it needs to be something unique to id each location. It could be address1 + address2 from location obj. Es-lint did not like index from map

File: components/Cards/SchoolCard/BackSchoolCard.js [hunk 34]
```diff
@@ -27,7 +30,7 @@ function BackSchoolCard({ cardFlipCallback, locations, logoSource, schoolName })
       <hr className={styles.divider} />
       <ul className={styles.locations}>
         {locations.map(location => (
-          <li key={`${JSON.stringify(location)}`}>
+          <li key={`${location.address1} ${location.address2}`}>
```
Comment (kylemh) [inline components/Cards/SchoolCard/BackSchoolCard.js]: 👍

File: components/Badge/__tests__/Badge.test.js [hunk 35]
```diff
@@ -17,8 +17,20 @@ describe('Badge', () => {
     createSnapshotTest(<Badge icon={badgeIcon} label="Badge Icon" className="test-class" />);
   });
 
-  it('should render the image after the label when `isImageFirst` is false', () => {
-    const wrapper = shallow(<Badge icon={badgeIcon} label="Badge Icon" isImageFirst={false} />);
-    expect(wrapper.children().first()).toContainExactlyOneMatchingElement('figure');
+  it('should render the image after the label when `isImageFirst` is true', () => {
```
Comment (kylemh) [inline components/Badge/__tests__/Badge.test.js]: Wonderful!

File: components/Cards/SchoolCard/FrontSchoolCard.css [hunk 36]
```diff
@@ -0,0 +1,65 @@
+.interactionsContainer {
+  align-items: center;
+  font-family: var(--primaryFontFamily), sans-serif;
```
Comment (kylemh) [inline components/Cards/SchoolCard/FrontSchoolCard.css]: I meant line 3 of a different file `SchoolCard/SchoolCard.css`

Basically, that should make the font choice for everything in the card.

File: components/Cards/SchoolCard/BackSchoolCard.css [hunk 37]
```diff
@@ -0,0 +1,18 @@
+.disclaimer {
+  font-size: 0.825rem;
+}
+
+.divider {
+  border: 1px solid #ffffff;
```
Comment (kylemh) [inline components/Cards/SchoolCard/BackSchoolCard.css]: sorry to keep catching these after sweeps, but `var(--white);`

File: components/Cards/SchoolCard/__tests__/SchoolCard.test.js [hunk 38]
```diff
@@ -0,0 +1,75 @@
...
+    );
+    onClickMock = jest.fn();
+    wrapper = mount(componentInstance);
+  });
+
+  it('should render with required props', () => {
+    createShallowSnapshotTest(componentInstance);
+  });
+
+  it('should change state when showBackOfCard/showFrontOfCard function is called', () => {
```
Comment (kylemh) [inline components/Cards/SchoolCard/__tests__/SchoolCard.test.js]: You can delete this test now - you've added a better test covering the same thing on line 61.
